src/sidebar/services/annotations.js

```diff
@@ -126,6 +126,26 @@ export default function annotationsService(api, store) {
     });
   }
 
+  /**
+   * Create a new empty "page note" annotation and add it to the store. If the
+   * user is not logged in, open the `loginPrompt` panel instead.
+   */
+  function createPageNote() {
+    const topLevelFrame = store.mainFrame();
+    if (!store.isLoggedIn()) {
+      store.openSidebarPanel('loginPrompt');
+      return;
+    }
+    if (!topLevelFrame) {
+      return;
+    }
+    const pageNoteAnnotation = {
+      target: [],
+      uri: topLevelFrame.uri,
+    };
+    create(pageNoteAnnotation);
```

Comment:
I don't know if this is any improvement:

```suggestion
    const { uri } = store.mainFrame() ?? {};
    
    if (!store.isLoggedIn()) {
      store.openSidebarPanel('loginPrompt');
      return;
    }

    if (!uri) {
      return;
    }

    create({ target: [], uri });
```